### ISSUE TITLE
Creating a Mng interface to create new Exchanges

### ISSUE THREAD

**Issue by sskhiri** [roq-messaging/RoQ#21]
RoQ has been designed to be elastic and then to be integrated with a Service Management such as RightScale or Scalr. In order to make easy this integration we need to:
1. provide a mng interface to provision new Exchanges
2. provide a registering mechanism to register strategy pattern for the autoscaling ploicy and Service management integration
Indeed, if it is easier to hard code the auto scaling policy, it would be much more easier if we delegate this feature to the user by implementing an interface an registrating to the core system the hard coded policy. Without the registration, the system does not enable the auto-scaling features and then we let the door open to interface with an external service manager.

**Comment by sskhiri** [comment]
In a first iteration, the global config will be a class that maintain the information associated with the global MonitorManager. In next iterations we should have:
- Decoupled Monitor manager and config manager (different threads)
- Persistence management of the of the global config in order to restart the queues on the next startup
- Distribution and HA of the global configuration by using distributed cache such as INFINISPAN, in this way, we can even start many stateless Monitor Manager

**Comment by sskhiri** [comment]
Updating the code with Jehan's suggestion about the process builder. Now we do not use script and jar anymore to start new monitor or exchange process. However, @Jehan I add to correct few things from your code
ProcessBuilder pb = new ProcessBuilder("java", "-Djava.library.path="
            + System.getProperty("java.library.path"), "-cp", System.getProperty("java.class.path"),
            MonitorLauncher.class.getCanonicalName(), new  Integer(frontPort).toString(), new Integer(statPort).toString());
1. removed the space after -cp, this made the VM not recognize the arg (I suffered to find it !)
2. splitted the arguments passed to the vm, otherwise it was all send in the arg[0](yes this is strange)

Now that is OK, and I have updated the testLogicalQueue Unit test. In addition I corrected few bugs in the queue creation and removal.
Now we can create a queue by:
LogicalQFactory factory = factory = new LogicalQFactory(<configServerAddress>);
factory.createQueue(qName, host);

that's all !! and for removing a queue:
factory.removeQueue(qName);

Still need to update the API for letting add an exchange to a queue on a target server.
